### PR TITLE
bsdlib: backport fixes to bsdlib 0.6.1

### PR DESCRIPTION
**Pull request by evenl**
Backporting fixes:

86c5465d4a206eea525bca53a607bb2d51378ff0
debf11a4e487ddf3a831fc58bf74ee99ba8eeee3

Manifest PR: https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/2255

Signed-off-by: Even Falch-Larsen <even.falch-larsen@nordicsemi.no>

### CODE REVIEW

File: bsdlib/include/bsd_limits.h [hunk 1]
```diff
@@ -10,6 +10,14 @@
...
  * @{
  * @brief Upper and lower bound limits of BSD library.
+ *
+ * @details This module provides:
+ * -# A common entry point for limits specific system initializations.
+ * -# Limits specific parameters:
+ *         -# Maximum number of concurrent sockets in the system.
+ *         -# Maximum number of concurrent secure sockets in the system.
+ *         -# Maximum number AT sockets and subscriptions for AT events in the system.
+ *         -# ..
```
Comment (rlubos) [inline bsdlib/include/bsd_limits.h:20]: I don't think we should leave it like this.

File: bsdlib/include/nrf_errno.h [hunk 2]
```diff
@@ -17,45 +17,44 @@
...
-#define NRF_EADDRINUSE      48  /**< Address already in use */
-#define NRF_ENETDOWN        50  /**< Network is down */
-#define NRF_ENETUNREACH     51  /**< Network is unreachable */
-#define NRF_ENETRESET       52  /**< Connection aborted by network */
-#define NRF_ECONNRESET      54  /**< Connection reset by peer */
-#define NRF_EISCONN         56  /**< Transport endpoint is already connected */
-#define NRF_ENOTCONN        57  /**< Transport endpoint is not connected */
-#define NRF_ETIMEDOUT       60  /**< Connection timed out */
-#define NRF_ENOBUFS         105 /**< No buffer space available */
+#define NRF_EPERM            1  //< Operation not permitted
```
Comment (rlubos) [inline bsdlib/include/nrf_errno.h:20]: Why was the comment style changed specifically for this bugfix release?

File: bsdlib/include/nrf_socket.h [hunk 3]
```diff
@@ -694,47 +596,36 @@ typedef struct
...
-#define NRF_GNSS_AGPS_NEQUICK_REQUEST             2
-#define NRF_GNSS_AGPS_SYS_TIME_AND_SV_TOW_REQUEST 3
-#define NRF_GNSS_AGPS_POSITION_REQUEST            4
-#define NRF_GNSS_AGPS_INTEGRITY_REQUEST           5
-
-/**@brief AGPS notification data frame used by the GPS module to let the application know it needs new APGS data.
- */
 typedef struct
 {
-    uint32_t sv_mask_ephe; /**< Bit mask indicating the satellite PRNs for which the assistance GPS ephemeris data is needed. */
```
Comment (rlubos) [inline bsdlib/include/nrf_socket.h:712]: Was it intended to remove those descriptions?

**Comment by rlubos** [comment]
BTW, a general question to the audience, is v0.6.1.1 a valid version, shouldn't this be a v0.6.2 (with PATCH version bumped up)? In any case, please update the commit title, 0.6.1 is already in the tree.

File: bsdlib/include/bsd_limits.h (hunk 1 above)
Comment (evenl) [reply bsdlib/include/bsd_limits.h:20]: I am not sure why this ended up in the diff, when I look back in the source repo this file has always been like this. It has just been fixed, but I can just remove it so it looks like how the diff say the previous version looked.

File: bsdlib/include/bsd_platform.h [hunk 4]
```diff
@@ -9,6 +9,15 @@
  * @defgroup bsd_platform_ipc BSD Platform
  * @ingroup bsd_platform
  * @{
+ * @brief Platform Initialization and system resource usage.
```
Comment (evenl) [inline bsdlib/include/bsd_platform.h:12]: Same goes for this, I just remove it so it looks like it did in 0.6.1

File: bsdlib/include/nrf_errno.h (hunk 2 above)
Comment (evenl) [reply bsdlib/include/nrf_errno.h:20]: I came with one of the commits I cherry-picked. I can change it back.

**Comment by evenl** [comment]
The reason for 0.6.1.1 is that 0.6.2 is already released. I can change it to something else, but 0.6.1.1 was my best suggestion because of this.

**Comment by rlubos** [comment]
The versioning of bsdlib is up to your team, if you say 0.6.1.1 is fine, then I'm ok with it. Just please update the commit title, to clearly indicate we update to version 0.6.1.1.

**Comment by evenl** [comment]
Sorry, I didn't see that the origianl 1.2 release contained a documentation only commit also. I have included that commit now. The diff is smaller, I think there may have been a slight variation between what was commited to nrfxlib and was later backported to the orignal internal repo which is the reason for the diff on the .h files.

**Comment by carlescufi** [comment]
@rlubos please merge this once is ready and then https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/2255 once the SHA has been updated there.

**Comment by evenl** [comment]
It is ready, I am not sure why the build job is failing, but I don't think it's due to my changes.

**Comment by rlubos** [comment]
@thst-nordic Any idea what's wrong with the CI here?

**Comment by rlubos** [comment]
According to the discussion on Teams, we need to update `Jenkinsfile` on this branch, in similar way it's done for -nrf:
https://github.com/NordicPlayground/fw-nrfconnect-nrf/blob/v1.2-branch/Jenkinsfile#L9

@evenl Could you add a commit to your branch that updates that line?

**Comment by lemrey** [comment]
Updated Jenkins file

File: Jenkinsfile [hunk 5]
```diff
@@ -6,7 +6,7 @@
 // express written agreement with Nordic Semiconductor ASA.
 //
 
-@Library("CI_LIB") _
+@Library("CI_LIB@v1.2-branch") _
```
Comment (rlubos) [inline Jenkinsfile:9]: Shouldn't this be a separate commit?

Comment (lemrey) [reply Jenkinsfile:9]: Yes, that makes sense.

File: Jenkinsfile [hunk 6]
```diff
@@ -25,7 +25,7 @@ pipeline
   parameters {
     booleanParam(name: 'RUN_TESTS', description: 'if false skip testing', defaultValue: true)
     booleanParam(name: 'RUN_BUILD', description: 'if false skip building', defaultValue: true)
-    booleanParam(name: 'RUN_DOWNSTREAM', description: 'if false skip downstream jobs', defaultValue: true)
+    booleanParam(name: 'RUN_DOWNSTREAM', description: 'if false skip downstream jobs', defaultValue: false)
```
Comment (rlubos) [inline Jenkinsfile:28]: What about this change, was it consulted with Thomas?

Comment (thst-nordic) [reply Jenkinsfile:28]: yes, all good.

**Comment by lemrey** [comment]
@eibe85 could you please run some tests on this?
@b-gent could you please have a look at the changelog?

**Comment by thst-nordic** [comment]
we can merge this with the failing status: 'CI/Jenkins/NRFXLIB/NRF'
because it is being tested in 
Manifest PR: NordicPlayground/fw-nrfconnect-nrf#2255

**Comment by b-gent** [comment]
Approving, please check my commit and squash it.

**Comment by rlubos** [comment]
@b-gent Thanks.
@lemrey @evenl Please squash the doc review commit.

**Comment by rlubos** [comment]
@thst-nordic Something's still not right with the CI here. It tries to checkout current `fw-nrfconnect-nrf` master instead of the release branch, even though the `Jenkinsfile` was updated, hence the failures. Please have a look at convenience.
